### PR TITLE
Patch: expose trim ext{2,3,4} setting

### PR DESCRIPTION
**Pull request by builder555**
Change-type: patch

### CODE REVIEW

**Comment by net147** [comment]
Fixes https://github.com/balena-io/etcher/issues/3942

**Comment by net147** [comment]
> > Fixes #3942
> 
> Can you test if the issue is fixed? I put the setting back but didn’t get a chance to flash a raw image yet. @net147 https://github.com/balena-io/etcher/releases/tag/untagged-568286a262c6719031c0

I can test if you can provide a Windows build. The link above doesn't work.

**Comment by builder555** [comment]
> Both those links give a Github 404 error, which I guess means it's a link only available to Balena employees?

i think you're right. i don't have access to change permissions right now, but i uploaded it here: https://ufile.io/486m2h0t the link will expire in 7 days.

**Comment by net147** [comment]
> > Both those links give a Github 404 error, which I guess means it's a link only available to Balena employees?
> 
> i think you're right. i don't have access to change permissions right now, but i uploaded it here: https://ufile.io/486m2h0t the link will expire in 7 days.

Works fine for me. With trim enabled the SD card doesn't match the image. With trim disabled the SD card matches the image.

**Comment by builder555** [comment]
That’s good news, I’ll test on Etcher Pro and make a release

**Review by mcraa — APPROVED** [review]

File: lib/gui/app/components/settings/settings.tsx [hunk 1]
```diff
@@ -37,6 +37,10 @@ async function getSettingsList(): Promise<Setting[]> {
 			name: 'errorReporting',
 			label: i18next.t('settings.errorReporting'),
 		},
+		{
+			name: 'autoBlockmapping',
+			label: 'Trim ext{2,3,4} partitions before writing (raw images only)',
```
Comment (lurch) [inline lib/gui/app/components/settings/settings.tsx:42]: Looking at the lines above and below this change, should the label-text here have been using `i18next.t()` ? :man_shrugging:

Comment (builder555) [reply lib/gui/app/components/settings/settings.tsx:42]: good call @lurch

**Pull request merged**